### ISSUE TITLE
Some markdown syntax difference with GitBook

### ISSUE THREAD

**Issue by azu** [honkit/honkit#77]
GitBook(Legacy) support code syntax with following syntax

    `` ` ``

HonKit does not support it.

Workaround: 

    <code>`</code>

Probably, this changes is caused by `kramed` to `@honkit/markup-it`
https://github.com/GitbookIO/gitbook-markdown/commit/0981578985933b9e3acd24adcb7cabb7e2c7849b#diff-e8acc63b1e238f3255c900eed37254b8

## Related

https://github.com/asciidwango/js-primer/pull/1200

**Comment by azu** [comment]
Background:

- GitBook use https://github.com/GitbookIO/kramed
- GitBook switch to in [markup-it@2](https://github.com/GitbookIO/markup-it/tree/2.2.0)? in https://github.com/GitbookIO/gitbook/tree/4.x.x ?

**Comment by azu** [comment]
💡 Idea: HonKit's parser is not pluggable.

https://github.com/honkit/honkit/blob/025344d16113594e62864986d11e044cb9b8fec8/packages/honkit/src/parsers.js#L13

We can implment parser as plugin and We will resolve this issue.
User can select parser of GitBook 3.x or new parser(GitBook 4.x based).
`gitbook-plugin-uml` will require 3.x parser for full compatible #82

**Comment by azu** [comment]
## Proposal: Downgrade markdown parser to kramed

Current HonKit has some issues on Markdown parser that is based on `markup-it@2`.

We can get more compatibility with GitBook Legacy by downgrad markdown parser.
This is temporary action in v3.

Probably, next major version will change it.

### Background

- GitBook v3(stable) use https://github.com/GitbookIO/kramed
- GitBook v4(beta) use markup-it@2
- HonKit use [HonKit Markdown Parser](https://github.com/honkit/honkit/tree/master/packages/%40honkit/markdown) based on markup-it@ 2

### Incompatible with GitBook Legacy

Some plugin depended on kramed parser #82

- [gitbook-plugin-uml](https://github.com/vowstar/gitbook-plugin-uml)
- (Please tell us if you found)

### Incompatible with [CommonMark](https://commonmark.org/)

GitBook v4(beta) and [HonKit Markdown Parser](https://github.com/honkit/honkit/tree/master/packages/%40honkit/markdown)  can not parse [CommonMark](https://commonmark.org/) syntax correctly

For example,

    `` ` ``

and

````
~~~md
 ```mermaid
 graph TD;
   A-->B;
   A-->C;
   B-->D;
   C-->D;
 ```
~~~
````

`kramed` can parse these.
Maybe, It is `marked` based parser and it close to CommonMark behavior

### TODO

- Use [gitbook-markdown@1.3.2](https://github.com/GitbookIO/gitbook-markdown) or Create fork of [gitbook-markdown@1.3.2](https://github.com/GitbookIO/gitbook-markdown) as `@honkit/markdown-legacy`

```diff
- const markdownParser = require("@honkit/markdown");
+ const markdownParser = require("@honkit/markdown-legacy");
```

It will improve GitBook(legacy) compatible and It help you to smooth migrate from GitBook.

### Future of Markdown parser

In V4 #15 ?

I have two proposals.

- Pluggable parser plugin
- Rewrite markdown parser with existing markdown parser library

#### Pluggable parser plugin

Currently, user can not select parser as plugin.
Probably, we can do it.

So, HonKit user will select `@honkit/markdown` or `@honkit/markdown-legacy`, or own custom parser.
It includes asciidoc parser.

####  Rewrite markdown parser with existing markdown parser library

We want to get [CommonMark](https://commonmark.org/) compatible + GitBook Extension parser in the future.
I think, It is good that we use [remark](https://github.com/remarkjs/remark) or [Marked](https://github.com/markedjs/marked).
These Markdown parser is compatible with [CommonMark](https://commonmark.org/) and we can add custom sytntax like `# Title {#title-id}`

We can not get incentive from original Markdown parser like kramed or markup-it.

📝 Actually, [kramed](https://github.com/GitbookIO/kramed) is a fork of old [marked](https://www.npmjs.com/package/marked)

kramed did want to support  custom syntax like mathjax.
It was the reason for creating kramed.

> Why fork marked ?
>
> marked hasn't been evolving as much as it could be lately and due to our needs with GitBook, we need features such as robust mathjax support and want to strive closer to the rising kramdown standard.
> https://github.com/GitbookIO/kramed

But GitBook/HonKit has supported mathjax by [mathjax plugin](https://github.com/GitbookIO/plugin-mathjax).
In other words, We not need to extend the parser.

So, We want to get rid of full scratched markdown parser. It make maintaince cost.

📝 Custom parser that is used [remark](https://github.com/remarkjs/remark) or [Marked](https://github.com/markedjs/marked) is needed for backward compatible.

**Comment by azu** [comment]
I've tried this proposal in #85 
[gitbook-markdown@1.3.2](https://github.com/GitbookIO/gitbook-markdown) does not work on WIndows.
The parser is someting wrong.

**Comment by azu** [comment]
https://github.com/honkit/honkit/releases/tag/v3.5.1 releaed.
https://github.com/honkit/honkit/releases/tag/v3.5.1 use @honkit/markdown-legacy that is based on GitBook v3 parser.
